backend/app/controllers/projects.js

```diff
@@ -62,19 +62,34 @@ module.exports = {
 
                     var productId = _.first(_.map(product, 'id'));
                     var flags = 0;
+                    var workflowId = null;
+                    var stages = [];
                     if (productId) {
-                        var workflowId = yield thunkQuery(
+                        workflowId = yield thunkQuery(
                             Workflow.select(Workflow.id).from(Workflow).where(Workflow.productId.equals(productId))
                         );
+                        workflowId = _.first(_.map(workflowId, 'id'));
                         flags = yield thunkQuery(
                             'SELECT DISTINCT "Discussions"."questionId" FROM "Discussions" ' +
                             'JOIN "Tasks" on "Discussions"."taskId" = "Tasks"."id" WHERE ' +
                             '"Tasks"."productId" = ' + productId + ' AND "Discussions".' +
                             '"isResolve" = false GROUP BY "Discussions"."questionId"'
                         );
                         flags = flags.length;
+                        stages = yield thunkQuery(
+                            WorkflowSteps
+                                .select(
+                                    WorkflowSteps.star())
+                                .from(WorkflowSteps)
+                                .where(WorkflowSteps.workflowId.equals(workflowId))
+                                .and(WorkflowSteps.isDeleted.isNull())
+                        );
+                        for (var index = 0; index < stages.length; index++) {
+                            stages[index].userGroups = [];
```

Comment:
Why are we doing this?

Reply:
Because the front end blows up without something in userGroups and this is a quick-and-dirty fix rather than rewriting it. If you want it done on the front end, we should revert the front end merge.

Reply:
I see. Okay, can we make a ticket to maybe clean up later?

Reply:
I can do it now if you want, just revert the front end commit.

Reply:
I mean we can keep it, its not too much of a big deal. We'll be refactoring this stuff most likely anyway.

Reply:
Okay.